src/runCompletion.ts

```diff
@@ -48,3 +49,23 @@ function getMaxResults(): number {
 
   return MAX_NUM_RESULTS;
 }
+
+export type KnownLanguageType = keyof typeof languages;
+
+export function getLanguageFileExtension(
+  languageId: string
+): string | undefined {
+  return languages[languageId as KnownLanguageType];
+}
+
+export function getFileNameWithExtension(document: TextDocument): string {
+  const { languageId, fileName } = document;
+  let filenameWithExtension = fileName;
+  if (document.isUntitled) {
```

Comment:
I always prefer early returns, so:
```suggestion
  if (!document.isUntitled) {
       return filenameWithExtension;
   }
```